middleware/markdown/markdown.go

```diff
@@ -99,74 +70,67 @@ func (md Markdown) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 			continue
 		}
 
+		var dirents []os.FileInfo
 		fpath := r.URL.Path
 		if idx, ok := middleware.IndexFile(md.FileSys, fpath, md.IndexFiles); ok {
+			// We're serving a directory index file, which may be a markdown
+			// file with a template.  Let's grab a list of files this directory
```

Comment:
As templates can invoke content modifying actions, I do not think that this would be enough to generate a unique ETag header.